libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -228,26 +269,42 @@ struct AdaptiveOrderedByIntegerMetric : private boost::noncopyable {
     unsigned int lastMasterVersion;
     unsigned int numPops;
 
-    unsigned int sinceLastFix;
+    unsigned int popsLastFix;
     unsigned int slowPopsLastPeriod;
     unsigned int pushesLastPeriod;
-    unsigned int priosLastPeriod;
-
-    unsigned long pmodAllDeq;
-
     unsigned int popsFromSameQ;
-    unsigned int ctr;
-
-    Index maxPrioDiffLastPeriod;
+    struct {
+      size_t pmodAllDeq;
+      unsigned int priosLastPeriod;
+      unsigned int numUmerges;
+      Index maxPrioDiffLastPeriod;
+    } stats;
     Index minPrio;
     Index maxPrio;
     substrate::PaddedLock<Concurrent> lock;
 
+    void cleanup() {
+      popsLastFix        = 0;
+      slowPopsLastPeriod = 0;
+      pushesLastPeriod   = 0;
+
+      stats.priosLastPeriod       = 0;
+      stats.maxPrioDiffLastPeriod = 0;
+
+      minPrio = std::numeric_limits<Index>::max();
```

Comment:
This issue was present in the old version too, but we should use the max and min from the comparator here and not rely on the numeric_limits. A related question here though: does this scheduler even support prioritizing higher integer priorities instead of lower ones? Now that I think about it some more that may throw off some of the arithmetic elsewhere. Fortunately we're not relying on this feature right now, so it's not something we have to fix immediately.

Reply:
From what I understand, this doesn't matter or have anything to do with ascending/descending - the minPrio and maxPrio are used to compute the priority range encountered in a period, which is later used to calculate the average items per priority and perform merging based on the calculation.

Reply:
Got it. That makes sense. Thanks for the clarification.